odoo/tests/common.py

```diff
@@ -2039,6 +2040,15 @@ def _get_modifier(self, field, modifier, default=False, modmap=None, vals=None):
                         # field value should be [(6, _, ids)], we want just the ids
                         field_val = field_val[0][2] if field_val else []
 
+                    # in an embedded view, the modifiers' view is not self._view so the fields for the domains cannot be
+                    # found but we have the field name and it's value, so we can guess a m2m field in a hack-ish way
+                    if f_['type'] is None and f.endswith('_ids') and len(field_val) == 1:
```

Comment:
I wouldn't rely on some heuristics about the field name.
Instead, you could check the type of the value itself
```suggestion
                    if f_['type'] is None and isinstance(field_val, list) and len(field_val) == 1:
```

Reply:
Note that non stored computed binary fields are sometimes used to store python structures such as `dict` or `list`, so this is not foolproof either

Reply:
I decided to neglect the case when the field name does not end with `_ids` at first, because there are *very* few of them, but now I think they are important ones; also I first thought it is the responsibility of the developer to follow the guidelines, but now I think it is not robust.
So I suggest just using `try`:
```python
 f_ = self._view['fields'].get(f, {'type': None})
try:
    if  f_['type'] is len(field_val)==1 and len(field_val[0]==3) and field_val[0][0]=Command.SET:
        field_val = field_val[0][2]

except (IndexError, TypeError, KeyError): # or just Exception ?)
    pass
```